[refresh2020q1][serf] - IO error while building serf

This error is occuring during the refresh process. I have posted the context of the error and linked the file at the bottom of the issue.

Error:

```
1.6.1/20200324181821/include/apr-1 -I/hab/pkgs/core/expat/2.2.7/20200324033915/include -I/hab/pkgs/core/openssl/1.0.2t/20200324035851/include buckets/deflate_buckets.c
buckets/deflate_buckets.c:18:10: fatal error: zlib.h: No such file or directory
   18 | #include <zlib.h>
      |          ^~~~~~~~
compilation terminated.
scons: *** [buckets/deflate_buckets.o] Error 1
scons: building terminated because of errors.
   serf: Build time: 0m4s
   serf: Exiting on error

```

Steps to reproduce:

```
cd /tmp/
git clone --depth 1 -b refresh/2020q1_008 --single-branch https://github.com/habitat-sh/core-plans.git /tmp/core-plans"
wget https://gist.githubusercontent.com/MindNumbing/255ca89882213bf0a4422f28fb902570/raw/b00ed3638b73d8a697ca411ea5154bd77118d456/hab_studio.env
source hab.env
hab studio enter
build core-plans/serf
```

Full Error Log:
[serf.txt](https://github.com/habitat-sh/core-plans/files/4458559/serf.txt)

--- a/serf/pass-env-to-scons.patch
+++ b/serf/pass-env-to-scons.patch
@@ -1,12 +1,12 @@
---- SConstruct.orig	2016-07-15 14:13:44.447114953 +0000
-+++ SConstruct	2016-07-15 14:14:54.744356018 +0000
-@@ -149,6 +149,9 @@
+diff --git a/SConstruct b/SConstruct
+index 211a1e3..e2cf623 100644
+--- a/SConstruct
++++ b/SConstruct
+@@ -176,6 +176,7 @@ if sys.platform == 'win32':
  env = Environment(variables=opts,
                    tools=('default', 'textfile',),
                    CPPPATH=['.', ],
-+                  ENV=os.environ,
-+                  CFLAGS=os.environ['CFLAGS'],
-+                  LINKFLAGS = os.environ['LDFLAGS'],
++                  ENV=os.environ
                    )
-
- env.Append(BUILDERS = {
+ 
+ gen_def_script = env.File('build/gen_def.py').rstr()

--- a/serf/plan.sh
+++ b/serf/plan.sh
@@ -35,7 +35,10 @@ do_build() {
     PREFIX="$pkg_prefix" \
     APR="$(pkg_path_for apr)" \
     APU="$(pkg_path_for apr-util)" \
-    OPENSSL="$(pkg_path_for openssl)"
+    ZLIB="$(pkg_path_for zlib)" \
+    OPENSSL="$(pkg_path_for openssl)" \
+    CFLAGS="$CFLAGS" \
+    LINKFLAGS="$LDFLAGS"
 }
 
 # Currently errors out on a comment style in a test suite, of all things.

